File: ecdpcode.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include <mmsystem.h>
#pragma hdrstop
#include "ecdpcode.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;

void device_setup()
{
 Form1->MediaPlayer1->Visible=false;
 Form1->MediaPlayer1->DeviceType=dtCDAudio;
 Form1->MediaPlayer1->TimeFormat=tfTMSF;
 Form1->MediaPlayer1->Notify=false;
 Form1->MediaPlayer1->AutoOpen=false;
 Form1->MediaPlayer1->Open();
}

void window_setup()
{
 Form1->Font->Name="Tahoma";
 Form1->Font->Size=14;
 Form1->BorderStyle=bsDialog;
}

void interface_setup()
{
 Form1->Caption="EASY CD PLAYER 0.8.2";
 Form1->Button1->Caption="Play";
 Form1->Button2->Caption="Pause";
 Form1->Button3->Caption="Next";
 Form1->Button4->Caption="Back";
 Form1->Button5->Caption="Eject";
 Form1->Label1->Caption="Tape:";
 Form1->Label5->Caption="Elapsed time:";
}

void label_setup()
{
 Form1->Label3->Caption="/";
 Form1->Label2->Caption=" ";
 Form1->Label4->Caption=" ";
 Form1->Label7->Caption=":";
 Form1->Label6->Caption=" ";
 Form1->Label8->Caption=" ";
}

void timer_setup()
{
 Form1->Timer1->Interval=1000;
 Form1->Timer1->Enabled=false;
}

void setup()
{
 device_setup();
 window_setup();
 interface_setup();
 label_setup();
 timer_setup();
}

void show_track()
{
 unsigned long int track;
 track=MCI_TMSF_TRACK(Form1->MediaPlayer1->Position);
 Form1->Label2->Caption=IntToStr(track);
 Form1->Label4->Caption=IntToStr(Form1->MediaPlayer1->Tracks);
}

void show_time()
{
  unsigned long int current;
  current=MCI_TMSF_MINUTE(Form1->MediaPlayer1->Position);
  Form1->Label6->Caption=IntToStr(current);
  current=MCI_TMSF_SECOND(Form1->MediaPlayer1->Position);
  Form1->Label8->Caption=IntToStr(current);
}

void do_play()
{
 Form1->Timer1->Enabled=true;
 Form1->MediaPlayer1->Play();
}

void do_pause()
{
 Form1->Timer1->Enabled=false;
 Form1->MediaPlayer1->Pause();
}

void do_eject()
{
 Form1->Timer1->Enabled=false;
 Form1->MediaPlayer1->Stop();
 Form1->MediaPlayer1->Eject();
}

void go_next()
{
 Form1->Timer1->Enabled=true;
 Form1->MediaPlayer1->Next();
 Form1->MediaPlayer1->Play();
}

void go_back()
{
 Form1->Timer1->Enabled=true;
 Form1->MediaPlayer1->Previous();
 Form1->MediaPlayer1->Previous();
 Form1->MediaPlayer1->Play();
}

//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
		: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormCreate(TObject *Sender)
{
 setup();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button1Click(TObject *Sender)
{
 do_play();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button2Click(TObject *Sender)
{
 do_pause();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button3Click(TObject *Sender)
{
 go_next();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button4Click(TObject *Sender)
{
 go_back();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button5Click(TObject *Sender)
{
 do_eject();
 label_setup();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormClose(TObject *Sender, TCloseAction &Action)
{
 MediaPlayer1->Stop();
 MediaPlayer1->Close();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Timer1Timer(TObject *Sender)
{
 show_track();
 show_time();
}
//---------------------------------------------------------------------------

File: ecdpcode.pas
unit ecdpcode;

{
 This sofware was made by Popov Evgeniy Alekseyevich.
 It is distributed under the GNU GENERAL PUBLIC LICENSE (Version 2 or higher).
}

interface

uses Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms, Dialogs, StdCtrls, MPlayer, ExtCtrls, mmsystem;

type
  TMainWindow = class(TForm)
    PlayButton: TButton;
    PauseButton: TButton;
    NextButton: TButton;
    BackButton: TButton;
    EjectButton: TButton;
    MusicStatus: TLabel;
    Timer: TTimer;
    CDPlayer: TMediaPlayer;
    procedure FormCreate(Sender: TObject);
    procedure PlayButtonClick(Sender: TObject);
    procedure PauseButtonClick(Sender: TObject);
    procedure NextButtonClick(Sender: TObject);
    procedure BackButtonClick(Sender: TObject);
    procedure EjectButtonClick(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure TimerTimer(Sender: TObject);
  private
    procedure device_setup();
    procedure window_setup();
    procedure interface_setup();
    procedure timer_setup();
    procedure setup();
  public
    { Public declarations }
  end;

var MainWindow: TMainWindow;

implementation

function show_music_status(const Position:LongWord): string;
var minutes,seconds,track:LongWord;
begin
 minutes:=MCI_TMSF_MINUTE(Position);
 seconds:=MCI_TMSF_SECOND(Position);
 track:=MCI_TMSF_TRACK(Position);
 Result:=IntToStr(track)+' - '+IntToStr(minutes)+':'+IntToStr(seconds);
end;

{$R *.dfm}

procedure TMainWindow.device_setup();
begin
 Self.CDPlayer.DeviceType:=dtCDAudio;
 Self.CDPlayer.TimeFormat:=tfTMSF;
 Self.CDPlayer.Visible:=False;
 Self.CDPlayer.Notify:=False;
 Self.CDPlayer.AutoOpen:=False;
 Self.CDPlayer.Open();
end;

procedure TMainWindow.window_setup();
begin
 Application.Title:='Easy CD player';
 Self.Caption:='Easy CD player 0.9.2';
 Self.Font.Assign(Screen.MenuFont);
 Self.Font.Size:= 14;
 Self.BorderStyle:=bsDialog;
end;

procedure TMainWindow.interface_setup();
begin
 Self.PlayButton.Caption:='Play';
 Self.PauseButton.Caption:='Pause';
 Self.NextButton.Caption:='Next';
 Self.BackButton.Caption:='Back';
 Self.EjectButton.Caption:='Eject';
 Self.MusicStatus.Caption:='';
end;

procedure TMainWindow.timer_setup();
begin
 Self.Timer.Interval:=1000;
 Self.Timer.Enabled:=false;
end;

procedure TMainWindow.setup();
begin
 Self.device_setup();
 Self.window_setup();
 Self.interface_setup();
 Self.timer_setup();
end;

procedure TMainWindow.PlayButtonClick(Sender: TObject);
begin
 Self.Timer.Enabled:=True;
 Self.CDPlayer.Play();
end;

procedure TMainWindow.PauseButtonClick(Sender: TObject);
begin
 Self.Timer.Enabled:=False;
 Self.CDPlayer.Pause();
end;

procedure TMainWindow.NextButtonClick(Sender: TObject);
begin
 Self.Timer.Enabled:=True;
 Self.CDPlayer.Next();
 Self.CDPlayer.Play();
end;

procedure TMainWindow.BackButtonClick(Sender: TObject);
begin
 Self.Timer.Enabled:=True;
 Self.CDPlayer.Previous();
 Self.CDPlayer.Play();
end;

procedure TMainWindow.EjectButtonClick(Sender: TObject);
begin
 Self.Timer.Enabled:=False;
 Self.CDPlayer.Stop();
 Self.CDPlayer.Eject();
 Self.MusicStatus.Caption:='';
end;

procedure TMainWindow.FormClose(Sender: TObject; var Action: TCloseAction);
begin
 Self.CDPlayer.Stop();
 Self.CDPlayer.Close();
end;

procedure TMainWindow.FormCreate(Sender: TObject);
begin
 Self.setup();
end;

procedure TMainWindow.TimerTimer(Sender: TObject);
begin
 Self.MusicStatus.Caption:=show_music_status(Self.CDPlayer.Position);
end;

end.